src/Cookie.php

```diff
@@ -207,12 +207,12 @@ public function domain_matches($domain) {
 			return false;
 		}
 
-		if (substr($domain, -1 * strlen($cookie_domain)) !== $cookie_domain) {
+		if (substr($domain, -(strlen($cookie_domain))) !== $cookie_domain) {
 			// The cookie domain should be a suffix of the passed domain.
 			return false;
 		}
 
-		$prefix = substr($domain, 0, strlen($domain) - strlen($cookie_domain));
+		$prefix = substr($domain, 0, -strlen($cookie_domain));
```

Comment:
Why is one occurrence of `strlen()` being put into parentheses while the other isn't?

Also, it might be good to calculate the `$prefix` only once.

Reply:
NVM the second remark, it's not the same `substr()`.

Reply:
I've added an extra commit which stores the calculated value and then automatically gets rid of the extra set of parentheses as well.